ts/src/scenario.ts

```diff
@@ -89,27 +113,84 @@ export class Scenario {
     return conductor;
   }
 
+  /**
+   * Create conductors with agents and add them to the scenario.
+   *
+   * The specified number of conductors is created and one agent is
+   * generated on each conductor.
+   *
+   * @param amount - The number of players to be created.
+   * @param networkConfig - Optional {@link NetworkConfig}
+   * @returns An array of {@link Player}s
+   */
+  async addPlayers(
+    amount: number,
+    networkConfig?: NetworkConfig,
+  ): Promise<Player[]> {
+    await this.ensureLocalServices();
+    return Promise.all(
+      new Array(amount).fill(0).map(async () => {
+        const conductor = await this.addConductor();
+        if (networkConfig) {
+          conductor.setNetworkConfig(networkConfig);
+        }
+        const agentPubKey = await conductor.adminWs().generateAgentPubKey();
+        return { conductor, agentPubKey };
+      }),
+    );
+  }
+
+  /**
+   * Installs the provided apps for the provided players.
+   *
+   * The number of players must be at least as high as the number of apps.
+   * **The agent pub key of the app options will be overwritten by the player's
+   * agent pub key**.
+   *
+   * @param appsWithOptions - The apps with options to be installed
+   * @param players - The players the apps are installed for
+   * @returns An array of player apps.
+   */
+  async installAppsForPlayers(
+    appsWithOptions: AppWithOptions[],
+    players: Player[],
+  ) {
+    await this.ensureLocalServices();
+    return Promise.all(
+      appsWithOptions.map((appWithOptions, i) => {
+        const player = players[i];
+        appWithOptions.options = appWithOptions.options ?? {};
+        appWithOptions.options.agentPubKey = player.agentPubKey;
```

Comment:
This silent overriding of the options here seems problematic to me. I think it should throw an error if an agent key is defined in the options and does not match the agent key of the player, otherwise one can end up in the wrong belief that the agent key in the option is being respected.

Reply:
It's mentioned in the doc comment that it'll be overwritten. https://github.com/holochain/tryorama/pull/280/files#diff-345008ae18189b3a070fa40c96e3c68f34c1140f13e589d73b89b68480eb126fR147

But fine, throwing an error is clearer.

Reply:
done https://github.com/holochain/tryorama/pull/280/commits/6c39340ef896b339368a56bedb232cbf36752a92#diff-345008ae18189b3a070fa40c96e3c68f34c1140f13e589d73b89b68480eb126fR160